electron/main/buffer.js

```diff
@@ -3,21 +3,39 @@ import { join } from "path"
 import { app } from "electron"
 import CONFIG from "../config"
 import { isDev } from "../detect-platform"
+import os from "node:os"
+import * as jetpack from "fs-jetpack"
 
+const untildify = (pathWithTilde) => {
+    const homeDirectory = os.homedir();
+    return homeDirectory
+      ? pathWithTilde.replace(/^~(?=$|\/|\\)/, homeDirectory)
+      : pathWithTilde;
+}
+
+const realpathSync = (path) => {
+  try {
+    return fs.realpathSync(path);
+  } catch (err) {
+    if (err.code !== "ENOENT") {
+      throw err;
+    }
+  }
+  return "";
+};
 
-export function getBufferFilePath() {
+export function getBufferFilePath(path) {
     let defaultPath = app.getPath("userData")
-    let configPath = CONFIG.get("settings.bufferPath")
-    let bufferPath = configPath.length ? configPath : defaultPath
-    let bufferFilePath = join(bufferPath, isDev ? "buffer-dev.txt" : "buffer.txt")
-    try {
-        // use realpathSync to resolve a potential symlink
-        return fs.realpathSync(bufferFilePath)
-    } catch (err) {
-        // realpathSync will fail if the file does not exist, but that doesn't matter since the file will be created
-        if (err.code !== "ENOENT") {
-            throw err
-        }
-        return bufferFilePath
+    let configPath = path ? path : CONFIG.get("settings.bufferPath")
+    // Expand the given path, preferring the config path to the default.
+    let bufferPath = untildify(configPath ? configPath : defaultPath)
+    let exists = jetpack.exists(bufferPath);
+    // Use the given path if it's an existing file or we will create a new one.
+    if (exists === "file" || exists === false) {
+      return bufferPath;
+    } else if (exists === "dir") { // Use the default file name if a directory is given.
+      return join(bufferPath, isDev ? "buffer-dev.txt" : "buffer.txt")
+    } else if (exists === "other") { // Follow the symlink
+      return realpathSync(bufferPath)
```

Comment:
I don't see the point of having a call to a local `realpathSync()` function here instead of just inlining the code. 

If I'm not missing something, with the current implementation  the local `realpathSync()` has a code path where it returns `""` that could never occur (or if it can occur, that seems also wrong, since it would result in `getBufferFilePath()` returning `""`)

Reply:
Ah, I was planning to add `realpathSync()` back to the fs-jetpack library at some point